include/zephyr/net/http/server.h

```diff
@@ -0,0 +1,175 @@
+/*
+ * Copyright (c) 2023, Emna Rekik
+ * Copyright (c) 2024 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#ifndef ZEPHYR_INCLUDE_NET_HTTP_SERVER_H_
+#define ZEPHYR_INCLUDE_NET_HTTP_SERVER_H_
+
+#include <stdint.h>
+
+#include <zephyr/kernel.h>
+#include <zephyr/net/http/parser.h>
+#include <zephyr/net/http/hpack.h>
+#include <zephyr/net/socket.h>
+
+#define HTTP_SERVER_CLIENT_BUFFER_SIZE CONFIG_HTTP_SERVER_CLIENT_BUFFER_SIZE
+#define HTTP_SERVER_MAX_STREAMS        CONFIG_HTTP_SERVER_MAX_STREAMS
+#define HTTP_SERVER_MAX_CONTENT_TYPE_LEN CONFIG_HTTP_SERVER_MAX_CONTENT_TYPE_LENGTH
+
+#define HTTP2_PREFACE "PRI * HTTP/2.0\r\n\r\nSM\r\n\r\n"
+
+enum http_resource_type {
+	HTTP_RESOURCE_TYPE_STATIC,
+	HTTP_RESOURCE_TYPE_DYNAMIC,
+	HTTP_RESOURCE_TYPE_REST
```

Comment:
Do we have anything differentiating REST and DYNAMIC right now? I suspect that they might almost be synonymous at this point - REST endpoints are mostly used for dynamic APIs.

The only other thing that I could anticipate maybe falling under the category of DYNAMIC is serving files via a filesystem, which would likely need to be somewhat dynamic in terms of filesystem operations.

Reply:
Good point. I think the REST was left here by mistake, I removed the functions dealing with REST already earlier but seem to have missed this enum.